src/overlays/actors/ovl_En_Invadepoh/z_en_invadepoh.c

```diff
@@ -17,495 +242,5248 @@ ActorInit En_Invadepoh_InitVars = {
+    EnInvadepoh_Alien_PathUpdate(this, play);
+    EnInvadepoh_Alien_StepYawAlongPath(this, 0x320, 0);
+
+    if (sAlienStateFlags[EN_INVADEPOH_GET_INDEX(&this->actor)] & ALIEN_STATE_FLAG_ACTIVE) {
+        Actor_SetScale(&this->actor, 0.01f);
+        EnInvadepoh_SnapToFloor(this);
+        EnInvadepoh_SpawnWarpEffect(&this->actor.world.pos);
+        EnInvadepoh_Alien_SetupWarpIn(this);
+    }
+}
+
+void EnInvadepoh_Alien_SetupWaitToRespawn(EnInvadepoh* this) {
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.base.acFlags &= ~AC_ON;
+    this->collider.base.ocFlags1 &= ~OC1_ON;
+    this->actor.flags &= ~ACTOR_FLAG_80000000;
+    this->alpha = 0;
+    this->actor.draw = NULL;
+    this->shouldDraw = false;
+    this->shouldDrawDeathFlash = false;
+    this->eyeBeamAlpha = 0;
+    this->actionFunc = EnInvadepoh_Alien_WaitToRespawn;
+}
+
+/**
+ * Waits until the current time is later than the alien's respawn time, then makes the alien warp in.
+ */
+void EnInvadepoh_Alien_WaitToRespawn(EnInvadepoh* this, PlayState* play) {
+    EnInvadepoh_Alien_PathComputeProgress(this);
+    EnInvadepoh_Alien_PathUpdate(this, play);
+    EnInvadepoh_Alien_StepYawAlongPath(this, 0x320, 0);
+
+    if (this->pathProgress > 0.0f) {
+        Actor_SetScale(&this->actor, 0.01f);
+        EnInvadepoh_SnapToFloor(this);
+        EnInvadepoh_SpawnWarpEffect(&this->actor.world.pos);
+        EnInvadepoh_Alien_SetupWarpIn(this);
+    }
+}
+
+void EnInvadepoh_Alien_SetupWarpIn(EnInvadepoh* this) {
+    Animation_MorphToLoop(&this->skelAnime, &gAlienFloatAnim, -6.0f);
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.base.acFlags &= ~AC_ON;
+    this->collider.base.ocFlags1 &= ~OC1_ON;
+    this->alpha = 0;
+    this->actor.draw = EnInvadepoh_Alien_Draw;
+    this->shouldDraw = true;
+    this->shouldDrawDeathFlash = false;
+    this->eyeBeamAlpha = 0;
+    this->actor.flags |= ACTOR_FLAG_80000000;
+    this->actionFunc = EnInvadepoh_Alien_WarpIn;
+}
+
+/**
+ * Slowly fade in the alien and its eye beams as it floats forward along its path. When the alien is halfway done with
+ * fading in, its collider becomes enabled, allowing the player to hit it and allowing it to hit the player.
+ */
+void EnInvadepoh_Alien_WarpIn(EnInvadepoh* this, PlayState* play) {
+    EnInvadepoh_Alien_PathComputeProgress(this);
+    EnInvadepoh_Alien_PathUpdate(this, play);
+    EnInvadepoh_Alien_StepYawAlongPath(this, 0x320, 0);
+    Actor_PlaySfx_Flagged(&this->actor, NA_SE_EN_FOLLOWERS_BEAM_PRE - SFX_FLAG);
+
+    if (this->pathProgress >= 0.9999f) {
+        this->pathCompleted = true;
+    }
+
+    if (this->alpha >= (255 - 6)) {
+        this->alpha = 255;
+    } else {
+        this->alpha += 6;
+    }
+
+    if (this->alpha > 128) {
+        this->collider.base.atFlags |= AT_ON;
+        this->collider.base.acFlags |= AC_ON;
+        this->collider.base.ocFlags1 |= OC1_ON;
+    }
+
+    if (this->alpha == 255) {
+        if (this->eyeBeamAlpha >= 245) {
```

Comment:
```suggestion
        if (this->eyeBeamAlpha >= (255 - 10)) {
```
You do something similar just above in the same function